packages/rabbitmq/src/rabbitmqTxSubmitProvider.ts

```diff
@@ -106,7 +113,7 @@ export class RabbitMqTxSubmitProvider implements TxSubmitProvider {
   async submitTx(signedTransaction: Uint8Array) {
     try {
       await this.#ensureQueue();
-      this.#channel!.sendToQueue(queue, Buffer.from(signedTransaction));
+      this.#channel!.sendToQueue(TX_SUBMISSION_QUEUE, Buffer.from(signedTransaction));
     } catch (error) {
       throw Cardano.util.asTxSubmissionError(error) || new Cardano.UnknownTxSubmissionError(error);
```

Comment:
Can we ever get a TxSubmissionError here?

Should we wait for actual submission to happen before resolving? :thinking: Ideally `submitTx` would resolve/reject when it's actually submitted or rejected by the underlying provider in worker.

Reply:
I confirm the doubt it seems you have, @mkazlauskas: ATM the `RabbitMqTxSubmitProvider` works in a _fire and forget_ fashion, that means
1. the `Promise` returned by the `submitTx` method resolves as soon as the RabbitMQ server have accepted the messages rather than waiting for an ACK from the worker;
2. the only reason why the `submitTx` method can raise an error is a connection with RabbitMQ error;
3. the `submitTx` caller has no way to have any feedback about the submission through the `RabbitMqTxSubmitProvider`, but needs to inspect the blockchain to gather such feedbacks.

This is not by my design decision, but by my low RabbitMQ knowledge 😃 writing that method was my really first interaction with RabbitMQ.

I leave to those who know the product better than me (you and @rhyslbw 😉) the decision to keep the _fire and forget_ behavior or changing it to wait for an ACK from the other side.

Reply:
> Ideally submitTx would resolve/reject when it's actually submitted or rejected by the underlying provider in worker.

Yes, that's dependent on the confirmation service. I'll create a ticket for that to follow-on